server/src/main/java/keywhiz/service/daos/SecretDAO.java

```diff
@@ -456,6 +517,13 @@ public ImmutableList<SecretSeriesAndContent> getSecretsBatched(int idx, int num,
     });
   }
 
+  public ImmutableList<SecretSeriesAndContent> getSecretsBatched(int idx, int num,
+      boolean newestFirst, Object principal) {
+      permissionCheck.checkAllowedOrThrow(principal, Action.READ, null);
+
+      return getSecretsBatched(idx, num, newestFirst);
+  }
```

Comment:
I'm wondering if, for clarity, we might want to have an override for checkAllowedOrThrow (and isAllowed) where we don't need to pass in a target. Reading the code outside of an IDE, the meaning of `null` in these calls isn't immediately obvious.

Reply:
That's a good point, I've added that in.